Microsoft.HealthDataAIServices stable api release 2024-09-20

chrisribe opened the pull request:
# ARM (Control Plane) API Specification Update Pull Request 

## Purpose of this PR

Creating a stable release: 2024-09-20 this is simply a clone of last version: 2024-02-28-preview
No changes where made to the API implementation.

## Review history
Private repo PR reference (approved):
https://github.com/Azure/azure-rest-api-specs-pr/pull/19485

BreakingChange-Python-Sdk signoff (fake warning comment):
https://github.com/Azure/azure-rest-api-specs-pr/pull/19485#issuecomment-2367453725


What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [x] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Convert existing [OpenAPI spec to TypeSpec spec](https://aka.ms/typespec/conversion) (do not combine this with implementing changes for a new API version).
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

chrisribe committed:
Microsoft.HealthDataAIServices stable api release 2024-09-20

chrisribe committed:
Merge branch 'main' into stableCtrlPlaneRelease

chrisribe commented:
@msyyc, could you please mark the 'BreakingChange-Python-Sdk' tag as a false positive and add the 'BreakingChange-Python-Sdk-Approved' tag, similar to what we did in the private repository?

Ref: https://github.com/Azure/azure-rest-api-specs-pr/pull/19485#issuecomment-2367453725
Thanks

chrisribe committed:
Merge branch 'main' into stableCtrlPlaneRelease

msyyc commented:
> @msyyc, could you please mark the 'BreakingChange-Python-Sdk' tag as a false positive and add the 'BreakingChange-Python-Sdk-Approved' tag, similar to what we did in the private repository?
> 
> Ref: [Azure/azure-rest-api-specs-pr#19485 (comment)](https://github.com/Azure/azure-rest-api-specs-pr/pull/19485#issuecomment-2367453725) Thanks

After `ARMSignedOff` label marked, SDK review will finish within one day.

msyyc commented:
![image](https://github.com/user-attachments/assets/f00b5508-cf65-465c-95d5-4528e594c8db)
There is Python SDK breakings detected however the details show nothing.
Add @raych1 for awareness.

msyyc committed:
Update readme.md

Python SDK could be generated from Typespec directly and no need to generate from Swagger.

msyyc commented on specification/healthdataaiservices/resource-manager/readme.md:
````diff
@@ -50,7 +59,6 @@ This is not used by Autorest itself.
 
 ```yaml $(swagger-to-sdk)
 swagger-to-sdk:
-  - repo: azure-sdk-for-python
````
Python SDK could be generated from Typespec directly and no need to generate from swagger.

msyyc committed:
Update readme.md

chrisribe committed:
Merge branch 'main' into stableCtrlPlaneRelease

raych1 commented:
> ![image](https://private-user-images.githubusercontent.com/70930885/374376941-f00b5508-cf65-465c-95d5-4528e594c8db.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg0MDY5MzAsIm5iZiI6MTcyODQwNjYzMCwicGF0aCI6Ii83MDkzMDg4NS8zNzQzNzY5NDEtZjAwYjU1MDgtY2Y2NS00NjVjLTk1ZDUtNDUyOGU1OTRjOGRiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDA4VDE2NTcxMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY0Yzk1NzQwOTM2YWJhMTI3ODg4NzgwY2VmNDMyOTMyMWVmYjJhZjFhYjkwYmNkNWYxOGM2NGRkNzg4OTE2ZTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.NSebBaaCmUS2K0KpbZslgrkfZPKealyVr0N_ElbSy-4) There is Python SDK breakings detected however the details show nothing. Add @raych1/ @wanlwanl for awareness.

@msyyc the breaking change was reported for the generated SDK based on swagger instead of typespec. SDK automation takes TypeSpec precedence over swagger and all the result will be overridden by the generation from TypeSpec, including SDK PR, breaking change rendering, etc.

chrisribe committed:
Merge branch 'main' into stableCtrlPlaneRelease
